dazel/lib/src/bazelify/build.dart

```diff
@@ -98,6 +97,11 @@ class BuildFile {
   /// - Some packages generate 1 or more dart_vm_binary or dart_web_application
   static Future<BuildFile> fromPackageDir(
       String packageDir, Pubspec pubspec, BuildConfigSet buildConfigs) async {
+    final packageName = pubspec.pubPackageName;
+    final buildConfig = buildConfigs.local.packageName == packageName
```

Comment:
How about overloading the `[]` operator to do this logic by default (prefer the local version).